sphinx/writers/html.py

```diff
@@ -511,10 +497,31 @@ def visit_literal(self, node: Element) -> None:
         if 'kbd' in node['classes']:
             self.body.append(self.starttag(node, 'kbd', '',
                                            CLASS='docutils literal notranslate'))
-        else:
+            return
+        lang = node.get("language", None)
+        if 'code' not in node['classes'] or not lang:
             self.body.append(self.starttag(node, 'code', '',
                                            CLASS='docutils literal notranslate'))
             self.protect_literal_text += 1
+            return
+
+        def warner(msg):
```

Comment:
Where is this helper function used? AFAIK, our highlighting module no longer support this argument.

Reply:
Removed.